### PR TITLE
Remove constructors for ov Exceptions

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - CVS-103685

### CODE REVIEW

File: src/inference/src/dev/iasync_infer_request.cpp [hunk 1]
```diff
@@ -193,9 +193,9 @@ void ov::IAsyncInferRequest::check_state() const {
     std::lock_guard<std::mutex> lock{m_mutex};
     switch (m_state) {
     case InferState::BUSY:
-        throw ov::Busy("Infer Request is busy");
+        ov::Busy::create("Infer Request is busy");
     case InferState::CANCELLED:
-        throw ov::Cancelled("Infer Request was canceled");
+        ov::Cancelled::create("Infer Request was canceled");
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/iasync_infer_request.cpp:198]:
```suggestion
        ov::Cancelled::throw("Infer Request was canceled");
```
?

File: src/core/builder/include/ngraph/builder/make_constant.hpp [hunk 2]
```diff
@@ -84,17 +84,17 @@ std::shared_ptr<Node> make_constant(const element::Type& type, const Shape& shap
...
+        OPENVINO_THROW("make_constant: Unsupported element type 'u1'");
     case element::Type_t::i4:
-        throw ngraph_error("make_constant: Unsupported element type 'i4'");
+        OPENVINO_THROW("make_constant: Unsupported element type 'i4'");
     case element::Type_t::u4:
-        throw ngraph_error("make_constant: Unsupported element type 'u4'");
+        OPENVINO_THROW("make_constant: Unsupported element type 'u4'");
     case element::Type_t::undefined:
-        throw ngraph_error("make_constant: Unsupported element type 'undefined'");
+        OPENVINO_THROW("make_constant: Unsupported element type 'undefined'");
```
Comment (ilya-lavrenov) [inline src/core/builder/include/ngraph/builder/make_constant.hpp:97]: can we merge all these cases throwing exceptions into a single one?

**Review by ilya-lavrenov — COMMENTED** [review]
Do we have number before and after on Linux with LTO?

**Comment by ilya-lavrenov** [comment]
> Do we have number before and after on Linux with LTO?

```
master:
-rwxrwxr-x 1 ilya ilya 15973208 4月  14 16:04 bin/intel64/Release/libopenvino.so.2023.0.0
exceptions:
-rwxrwxr-x 1 ilya ilya 15625048 4月  14 16:07 bin/intel64/Release/libopenvino.so.2023.0.0
```

File: src/inference/src/dev/iasync_infer_request.cpp (hunk 1 above)
Comment (ilyachur) [reply src/inference/src/dev/iasync_infer_request.cpp:198]: throw is a key word

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**